presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -93,7 +102,9 @@ public ContinuousTaskStatusFetcher(
             Duration maxErrorDuration,
             ScheduledExecutorService errorScheduledExecutor,
             RemoteTaskStats stats,
-            boolean isBinaryTransportEnabled)
+            boolean binaryTransportEnabled,
```

Comment:
Should we move codec and response handler to a class (TaskStatusRequest?) and pass it to  ContinuousTaskStatusFetcher? We can get rid of these if checks and casts.